LPC SPI speed increase

agansari opened the pull request:
Attaches SPI device to the main clock source, SPI speed on LPC devices is the same as core's speed.

agansari committed:
soc: lpc devices spi clock speed increase

High Speed Flexcomm device that handles high speed SPI transfer are
mapped to the same High Frequency clock that the ARM core uses.
This allows for higher frequency SPI traffic.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari committed:
drivers: spi_mcux_flexcomm clock increase

Attached Flexcomm SPI driver to the main clock used by the core.
This means setting the SPI clock the same value as the core clock.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

MaureenHelm reviewed (changes requested):
Please update `tests/drivers/spi/spi_loopback/boards/lpcxpresso*.conf` too

agansari committed:
tests: LPC platforms SPI use default frequncy

After increasing SPI speeds on LPC family SoCs, the slow clock
speeds have been removed and default clock speeds are in use now.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari commented:
> Please update `tests/drivers/spi/spi_loopback/boards/lpcxpresso*.conf` too

@MaureenHelm I've done so, please review it again.
Also tested on lpc55s69 and lpc54114, test passes.
